### PR TITLE
fix(core): character descenders overflow the line

### LINKED ISSUE #2374: 🐞 - Character descenders overflow the line
### Which @taiga-ui/* package(s) are the source of the bug?

core

### Please provide a link to a minimal reproduction of the bug

https://taiga-ui.dev/form (check first input)

### Is this issue blocking you?

Non-Blocking

### Description

Characters like "y", "p", "g" (i.e. which have descenders) are not fully visible in field placeholders (at least in `tui-input`).

Снимок экрана 2022-08-15 в 17 28 24

### Angular version

14.1.1

### Taiga UI version

2.58.1

### Which browsers have you used?

- [X] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### Which operating systems have you used?

- [X] macOS
- [ ] Windows
- [ ] Linux
- [ ] iOS
- [ ] Android

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the current behavior?

Closes #2374

## What is the new behavior?

image

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by waterplea** [comment]
I believe all you need to do here is to remove padding from `tui-value-decoration` and change its `height` and `line-height` to:
```css
height: 2.5rem;
line-height: 2.25;
```

**Comment by splincode** [comment]
@waterplea fixed))

**Review by nsbarsukov — APPROVED** [review]

**Comment by waterplea** [comment]
> @waterplea fixed))

What's the purpose of all the other changes?

File: projects/core/components/primitive-textfield/value-decoration/value-decoration.style.less [hunk 1]
```diff
@@ -1,14 +1,23 @@
...
+    margin: -1.15rem 0;
     box-sizing: content-box;
-    color: var(--tui-text-01);
+    color: var(--tui-text-03); // @note: override color of text ellipsis
     animation: tuiPresent 1s;
+    vertical-align: middle;
+    text-overflow: ellipsis;
+    white-space: nowrap;
+
+    &:after {
```
Comment (splincode) [inline projects/core/components/primitive-textfield/value-decoration/value-decoration.style.less:15]: @waterplea without it, it doesn't align to the baseline

File: projects/core/components/primitive-textfield/value-decoration/value-decoration.style.less [hunk 2]
```diff
@@ -1,14 +1,23 @@
...
-    line-height: 1;
+    height: 2.5rem;
+    line-height: 2.25;
     width: 100%;
-    margin: -1.25rem 0;
-    padding: 0.625rem 0;
+    margin: -1.15rem 0;
     box-sizing: content-box;
-    color: var(--tui-text-01);
+    color: var(--tui-text-03); // @note: override color of text ellipsis
```
Comment (splincode) [inline projects/core/components/primitive-textfield/value-decoration/value-decoration.style.less:9]: @waterplea without it, ellipsis color use black

File: projects/core/components/primitive-textfield/primitive-textfield.style.less [hunk 3]
```diff
@@ -111,3 +111,11 @@
     position: relative;
     pointer-events: none;
 }
+
+.t-value-decoration {
```
Comment (splincode) [inline projects/core/components/primitive-textfield/primitive-textfield.style.less:115]: @waterplea without it, the text ellipsis just appears, although the text of the t-filler is hidden, but the ellipsis is shown, is this a browser bug?

File: projects/core/components/primitive-textfield/value-decoration/value-decoration.style.less (hunk 2 above)
Comment (waterplea) [reply projects/core/components/primitive-textfield/value-decoration/value-decoration.style.less:9]: Got it

File: projects/core/components/primitive-textfield/value-decoration/value-decoration.style.less (hunk 1 above)
Comment (waterplea) [reply projects/core/components/primitive-textfield/value-decoration/value-decoration.style.less:15]: Got it

**Review by waterplea — APPROVED** [review]

File: projects/core/components/primitive-textfield/primitive-textfield.style.less (hunk 3 above)
Comment (waterplea) [reply projects/core/components/primitive-textfield/primitive-textfield.style.less:115]: Got it

**Pull request merged**